Add Apps (#2335)
balena-io/docs: pages/learn/manage/apps.md

Apps will form the basis for our multi-app implementation which we’re continuing to work towards. In the future instead of a many-to-one Fleet:App relationship, we’ll have a many-to-many relationship, permitting multiple Apps to be installed on multiple fleets.

## Creating Apps

If you’ve previously worked with fleets on balenaCloud, the process of creating an App and [deploying][deploying] a release is very similar.

In order to create your own App, the code currently needs to be in a GitHub repository. This is due to the fact that when the App is deployed, the latest code is pulled from GitHub. This will change in the future as and when we add support for private apps. The App has to include a balena.yml file in the root directory of the project. Here's a balena.yml file of balenaDash for reference:

```yaml
name: balenaDash
description: >-
  Raspberry Pi-powered digital signage or website viewer, remotely accessible
  from anywhere.
type: sw.application

> joinable: false

Author: Good catch

assets:
  repository:
    type: blob.asset
    data:
      url: 'https://github.com/balenalabs/balena-dash'
  logo:
    type: blob.asset
    data:
      url: >-
        https://raw.githubusercontent.com/balenalabs/balena-dash/master/assets/logo.png
data:
  applicationConfigVariables:
    - BALENA_HOST_CONFIG_gpu_mem: 128
    - BALENA_HOST_CONFIG_dtoverlay: "vc4-fkms-v3d"
  applicationEnvironmentVariables:
    - SHOW_CURSOR: 0
    - CONTROL_TV: 0
    - ENABLE_GPU: 0
    - KIOSK: 1
    - PERSISTENT: 1
    - PORTAL_SSID: balenaDash
    - PORTAL_PASSPHRASE: balenaDash
    - ACTIVITY_TIMEOUT: 600
    - LAUNCH_URL: ''
    - ROTATE_DISPLAY: normal
    - WINDOW_POSITION: '0,0'
    - LOCAL_HTTP_DELAY: 0
    - FBCP_DISPLAY: ''
  defaultDeviceType: raspberrypi4-64
  supportedDeviceTypes:
    - raspberrypi4-64
    - fincm3
    - raspberrypi3
    - raspberrypi3-64
    - intel-nuc
    - genericx86-64-ext
    - raspberrypi400-64
version: 2.0.0
```

> 

Author: Added!

> Next, navigate to the Apps tab in the sidebar and click the `Create App` button in the balenaCloud dashboard.
> 
> Fill the details for your App in the modal, and click `Create new app` button to create your App.
> 
> Next, we have to push a new release to your newly created App on balenaCloud. If you are new to this, then follow the Getting Started guide.

Reviewer: ```suggestion
Next, navigate to the Apps tab in the sidebar and click the `Create App` button in the balenaCloud dashboard. Fill the details for your App in the modal, and click `Create new app` button to create your App. Next, we need to push a new release to your newly created App on balenaCloud. If you are new to this, then follow the [Getting Started](https://balena.io/docs/learn/getting-started/raspberrypi3/nodejs/) guide.
```

> ## Sharing an App on balenaHub

Reviewer: ```suggestion
## Publishing an App on balenaHub
```

I like publishing more?

![Enable App visibility on balenaHub](/img/common/app/visibility-toggle.png)

Apps marked with visibility set to ‘on’ in the dashboard will be made available as another resource for the community on balenaHub. Users can deploy Apps from balenaHub to their own fleets managed in their own balenaCloud accounts using the `Deploy` button with each App. 

## Deploying Apps from balenaHub

![Deploy an App from balenaHub](/img/common/app/deploy-from-balenahub.png)

When the deploy button on balenaHub is used, the balenaCloud builder will fetch the latest source from the linked GitHub repository and create a new build for the chosen fleet. This process uses the [Deploy with balena](https://www.balena.io/docs/learn/deploy/deploy-with-balena-button/) workflow in order to guide users through the process of creating/signing in to a balenaCloud account, creating a fleet and deploying the App.

## Support access

> If you require assistance with an App and have contacted us for support, agents may ask you to ‘enable support access’. This can be achieved via toggling the `Support access` setting, found within the settings page for your App, to on. 

Reviewer: ```suggestion
If you require assistance with an App and have contacted balena for support, agents may ask you to ‘enable support access’ for the App. This can be achieved via toggling the `Support access` setting, found within the settings page to on. 
```

__Note:__ If the dropdown for the destination organization is empty and greyed out, ensure that you have created an empty App in the target organization with the same name as the source App, and that the user that is transferring ownership of the App from the source organization has been added as a Developer to the target App.

Once the ownership has been set to the new owner, the source App owner will no longer be a member of the target App. If required, you will need to invite them to become a member of the App again. All other members of the source App will retain their membership of the target App once the transfer is complete.

## The future roadmap

The functionality for Apps we have today is only the start. Apps added via the dashboard can be shared on balenaHub for others to deploy to their own fleets which serves to help others, and we’ll continue building out to expand to installing private Apps on private fleets, automatic updates, and eventually the ability to install multiple Apps on a single fleet. Keep an eye on [our blog][blogUrl] for all the latest news.

> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion

```
